README.md

```diff
@@ -39,7 +39,166 @@ or
+   - PostgreSQL 12.0
+   - PostgreSQL 11.2
+   - PostgreSQL 10.7
+   - PostgreSQL 9.6.12
+   - PostgreSQL 9.5.16
+   - PostgreSQL 9.4.21
+
+* **A connection to the master instance**. Log-based replication will only work by connecting to the master instance.
+
+* **wal2json plugin**: To use Log Based for your PostgreSQL integration, you must install the wal2json plugin.
+  The wal2json plugin outputs JSON objects for logical decoding, which the tap then uses to perform Log-based Replication.
+  Steps for installing the plugin vary depending on your operating system. Instructions for each operating system type
+  are in the wal2json’s GitHub repository:
+
+  * [Unix-based operating systems](https://github.com/eulerto/wal2json#unix-based-operating-systems)
+  * [Windows](https://github.com/eulerto/wal2json#windows)
+
+
+* **postgres config file**: Locate the database configuration file (usually `postgresql.conf`) and define
+  the parameters as follows:
+
+    ```
+    wal_level=logical
+    max_replication_slots=5
+    max_wal_senders=5
+    ```
+
+    Restart your PostgreSQL service to ensure the changes take effect.
+  
+    **Note**: For `max_replication_slots` and `max_wal_senders`, we’re defaulting to a value of 5.
+    This should be sufficient unless you have a large number of read replicas connected to the master instance.
+
+
+* **Existing replication slot**: Log based replication requires a dedicated logical replication slot.
+  In PostgreSQL, a logical replication slot represents a stream of database changes that can then be replayed to a
+  client in the order they were made on the original server. Each slot streams a sequence of changes from a single
+  database.
+  
+  Login to the master instance as a superuser and using the `wal2json` plugin, create a logical replication slot:
+  ```
+    SELECT *
+    FROM pg_create_logical_replication_slot('pipelinewise_<database_name>', 'wal2json');  
+  ```
+
+  **Note**: Replication slots are specific to a given database in a cluster. If you want to connect multiple
+  databases - whether in one integration or several - you’ll need to create a replication slot for each database.
+
+## To run tests:
+
+1. Install python test dependencies in a virtual env and run nose unit and integration tests
+```
+  python3 -m venv venv
+  . venv/bin/activate
+  pip install --upgrade pip
+  pip install .[test]
+```
+
+2. You need to have a postgres database to run the tests and export its credentials:
+```
+  export TAP_POSTGRES_HOST=<postgres-host>
+  export TAP_POSTGRES_PORT=<postgres-port>
+  export TAP_POSTGRES_USER=<postgres-user>
+  export TAP_POSTGRES_PASSWORD=<postgres-password>
+```
+
+Test objects will be created in the `postgres` database.
+
+3. To run the tests:
+```
+  pytest tests
+```
+
+### To run pylint:
+
+1. Install python dependencies and run python linter
+```
+  python3 -m venv venv
+  . venv/bin/activate
+  pip install --upgrade pip
+  pip install .
+  pip install pylint
```

Comment:
No need to install `pylint` , it will be installed with `pip install .[test]`

Reply:
fixed by 32ab70f084512d790a7e52077c21c0d7fb9e7b06